burnman/layer.py

```diff
@@ -0,0 +1,899 @@
+                          pressure_top=None, n_max_iterations=50, max_delta = 1.e-5):
+        """
+        Sets the pressure mode of the layer, which can either be 'user-defined', or 'self-consistent'.
+        
+        Parameters
+        ----------
+        pressure_mode : string
+        This can be set to 'user-defined' or 'self-consistent'
+        pressures : array of floats
+        Pressures [Pa] to set layer to ('user-defined'). This should be the same length as defined radii array for the layer
+        pressure_top : float
+        Pressure [Pa] at the top of the layer.
+        gravity_bottom : float
+        gravity [m/s^2] at the bottom the layer
+        n_max_iterations : int
+        Maximum number of iterations to reach self-consistent pressures (default = 50)
+        max_delta : float
+        Relative update to the highest pressure in the layer between iterations to stop iterations (default = 1.e-5)
+        
+        
+        Note
+        ---------
+        'user-defined' = fixes the pressures with the profile input by the user in the 'pressures' argument. 
+        'self-consistent' = user needs to give gravity_bottom [m/s^2] and pressure_top [Pa] to compute self-consistent pressures.
+        """
+        self.reset()
+        assert(pressure_mode == 'user-defined' or pressure_mode == 'self-consistent')
+        
+        self.pressure_mode= pressure_mode
+        self.gravity_bottom = gravity_bottom
+        
+        if pressure_mode == 'user-defined':
+            assert(len(pressures) == len(self.radii))
+            self.pressures = pressures
+            warnings.warn("By setting the pressures in Layer it is unlikely to be self-consistent")
+        
+        if pressure_mode == 'self-consistent':
+            self.pressure_top = pressure_top
+            self.n_max_iterations = n_max_iterations
+            self.max_delta = max_delta
+                          
+
+    def make( self):
+        """
+        This routine needs to be called before evaluating any properties. If pressures and temperatures are self-consistent, they
+        are computed here. Also initializes an array of materials to compute properties from.
+        """
+        self.reset()
+        if not hasattr(self,'material'):
+            raise AttributeError(' set_meterial() for layer before make()')
+        if not hasattr(self,'temperature_mode'):
+            raise AttributeError(' set_temperature_mode() for layer before make()')
+        if not hasattr(self,'pressure_mode'):
+            raise AttributeError(' set_pressure_mode() for layer before make() ')
+
+
+        if self.pressure_mode == 'user-defined':
+            self.temperatures = self._evaluate_temperature(
+                self._pressures, self.temperature_top)
+
+        if self.pressure_mode == 'self-consistent':
+            new_press = self.pressure_top + \
+                (-self.radii + max(self.radii)) * \
+                2.e5  # initial pressure curve guess
```

Comment:
2.e5 is a really big number. Is it possible to make a more reasonable guess? Otherwise lots of low pressure EoSes are going to fail here.